amoro-ams/amoro-ams-optimizer/amoro-optimizer-spark/src/main/java/org/apache/amoro/optimizer/spark/SparkOptimizerExecutor.java

```diff
@@ -68,7 +68,11 @@ protected OptimizingTaskResult executeTask(OptimizingTask task) {
       return result;
     } catch (Throwable r) {
       LOG.error(
-          "Optimizer executor[{}] executed task[{}] failed, and cost {}", threadName, task, r);
+          "Optimizer executor[{}] executed task[{}] failed, and cost {}",
+          threadName,
+          task.getTaskId(),
+          (System.currentTimeMillis() - startTime),
+          r);
```

Comment:
Exception stack will not be logged in this way.

Reply:
log4j2 supports this format and can correctly log the exception stack trace.